### ISSUE TITLE
$nor filter no longer works with __raw___

### ISSUE THREAD

**Issue by apendleton** [hmarr/mongoengine#560]
Upgrading from 0.6.20 caused a **raw** query that had worked in 0.6.19 to stop working.  The query looks like this:

```
Doc.objects(__raw__={'deleted': False, 'scraped': 'yes', '$nor': [{'views.extracted': 'no'},{'attachments.views.extracted':'no'}]})
```

and now throws an exception:

```
pymongo.errors.OperationFailure: database error: $and/$or/$nor must be a nonempty array
```

Clearly the $nor array is non-empty on the way in, so looks like it's being mangled somewhere.

**Comment by rozza** [comment]
Fixed in MongoEngine/mongoengine@2f6b1c761116c74560ce05ad9d2553d46fc2684a

**Comment by rozza** [comment]
Just an FYI - we're moving! Please report issues on: http://github.com/MongoEngine/mongoengine
